src/lorentz_boost/boost_parameter/boost_vector/beta.jl

```diff
@@ -71,7 +71,7 @@ struct BetaVector{T <: Real} <: AbstractBoostVector
         b2 = x^2 + y^2 + z^2
         b2 <= 1 || throw(
             InvalidInputError(
-                "wrong length of the beta vector ($x, $y, $z). Its length needs to be less or equal to one, but x^2 + y^2 + z^2 = $b2 is given.",
+                "wrong length of the beta vector, its length needs to be less or equal to one",
```

Comment:
As an explanation for why I had to remove the extra information here: It's fine to throw errors on the GPU, but the String interpolation is not static, so that becomes invalid IR. In this case here, since the check is entirely runtime dependent, there's no way to hide it away from the compiler. The only other choice might be again some form of global switch to enable/disable input checks like this one.

Some of the other string interpolations further below could maybe be hidden via dispatch, though if the check fails on the GPU, it will again fail to compile with cryptic "invalid IR" messages, instead of showing the intended error message, so I'm not sure if that's actually a good idea.

Reply:
Hm that's strange.

Reply:
As I said, for now it's fine, but we should think about a more sophisticated solution.